platform/security/idp/security-idp-server/src/test/groovy/org/codice/ddf/security/idp/server/IdpEndpointSpec.groovy

```diff
@@ -0,0 +1,475 @@
+        when: "a relaystate already exists, shouldn't start over"
+        idpEndpoint.cookieCache.cacheSamlAssertion(cookieVal, Mock(Element))
+        idpEndpoint.cookieCache.addActiveSp(cookieVal, sp1entityid)
+        idpEndpoint.logoutStates.encode(cookieVal, new LogoutState(sp1entityid as Set))
+
+        response = idpEndpoint.processRedirectLogout(samlRequest,
+                null,
+                relayState,
+                sigAlg,
+                sig,
+                request)
+
+        then:
+        (response.entity as String).contains("Logout already in progress")
+    }
+
+    def "process redirect logout initial with one other sp"() {
+        setup:
+        def relayState = "relayState"
+        def samlResponse = RestSecurity.deflateAndBase64Encode("b64deflatedsamlstring")
+        def sig = "mysignature"
+        def sigAlg = "signaturealgorithm"
+        def cookieVal = "cookievalue"
+        def nameId = "nameid"
+        def requestId = "originalRequestId"
+        def responseId = "originalResponseId"
+
+        and: "configure idpEndpoint state"
+        def targetSp = sp2entityid
+        idpEndpoint.cookieCache.cacheSamlAssertion(cookieVal, Mock(Element))
+        idpEndpoint.cookieCache.addActiveSp(cookieVal, targetSp)
+        def logoutState = new LogoutState([targetSp] as Set)
+        logoutState.with {
+            it.nameId = nameId
+            originalIssuer = sp1entityid
+            originalRequestId = requestId
+            initialRelayState = relayState
+        }
+        idpEndpoint.logoutStates.encode(cookieVal, logoutState)
+
+        idpEndpoint.logoutService = Mock(LogoutService) {
+            extractSamlLogoutResponse(_ as String) >> Mock(LogoutResponse) {
+                getIssuer() >> Mock(Issuer) {
+                    getValue() >> sp1entityid
+                }
+                getStatus() >> Mock(Status) {
+                    getStatusCode() >> Mock(StatusCode) {
+                        getValue() >> StatusCode.SUCCESS_URI
+                    }
+                }
+                getID() >> responseId
+            }
+            buildLogoutRequest(_ as String, _ as String) >> {String name, String target ->
+                return new LogoutServiceImpl().buildLogoutRequest(name, targetSp)
+            }
+        }
+
+        def request = Mock(HttpServletRequest) {
+            getRequestURL() >> new StringBuffer("destinationurl")
+            getCookies() >> {
+                [Mock(Cookie) {
+                    getName() >> Idp.COOKIE
+                    getValue() >> cookieVal
+                }]
+            }
+        }
+
+        when:
+        def response = idpEndpoint.processRedirectLogout(null,
+                samlResponse,
+                relayState,
+                sigAlg,
+                sig,
+                request)
+
+        then:
+        notThrown(Exception)
+    }
+
+    def "process redirect logout initial with one other sp partial logout"() {
+        setup:
```

Comment:
If it is worth the effort, consider moving this into a general setup method since other tests seem to use this as well